### PR TITLE
[#1829] engine: Delete split objects properly

### LINKED ISSUE #1829: Оbject was not deleted after 'neofs-cli control drop-objects' command successfully completed
## Expected Behavior
Object should be deleted

## Current Behavior
Object was not deleted

## Steps to Reproduce (for bugs)
1.Create container:
```
COMMAND: 'neofs-cli --config /tests/neofs_cli_configs/empty_passwd.yml container create --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8081' --wallet '/tests/TemporaryDir/860d6d63-e60a-46e6-bb1d-3d3f07b39dad.json' --await --policy "REP 1 CBF 1 SELECT 1 FROM * FILTER 'UN-LOCODE' EQ 'RU LED' AS LOC"'
OUTPUT:
 container ID: FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa
awaiting...
container has been persisted on sidechain
```
2.Put object:
```
02:11:40 [INFO]  [92mExecuting command: neofs-cli --config /tests/neofs_cli_configs/empty_passwd.yml object put --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8082' --wallet '/tests/TemporaryDir/860d6d63-e60a-46e6-bb1d-3d3f07b39dad.json' --cid 'FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa' --file '/tests/TemporaryDir/bcc8a209-5788-4f94-922e-cf59fa4a5d66' --no-progress [0m
02:11:41 [INFO]  [92mOutput: [/tests/TemporaryDir/bcc8a209-5788-4f94-922e-cf59fa4a5d66] Object successfully stored
  OID: E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU
  CID: FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa
```
3.Drop object
```
02:11:50 [INFO]  [92mExecuting command: neofs-cli control drop-objects  -o FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa/E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU --endpoint obj-nn-test-node2.spb.yadro.com:8184 --wallet /.setup/services/storage/wallet04.json --config /tests/neofs_cli_configs/empty_passwd.yml [0m
02:11:50 [INFO]  [92mOutput: Objects were successfully marked to be removed.
```
4.Then try to get object. It was successsfull
```
02:11:50 [INFO]  [92mExecuting command: neofs-cli --config /tests/neofs_cli_configs/empty_passwd.yml object get --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8082' --wallet '/tests/TemporaryDir/860d6d63-e60a-46e6-bb1d-3d3f07b39dad.json' --cid 'FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa' --oid 'E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU' --file 'TemporaryDir/2f05faa2-40aa-40f7-9811-e62cdc466c24' --no-progress [0m
02:11:51 [INFO]  [92mOutput: [TemporaryDir/2f05faa2-40aa-40f7-9811-e62cdc466c24] Object successfully saved
ID: E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU
CID: FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa
Owner: NPZAC6w7GehsCkii6g9pfuxhYpioLC3xSe
CreatedAt: 19
Size: 2000
HomoHash: 3296263dd3c7323b50cde1add2a1506d555323862ac2a5d9a3667bed8084775435a2abb6ac688ef2953dc34a22cfd72759abcdb0d4576e495859625f49187cd3
Checksum: 7f4dc5c06f7c6bf86b939cc86875d7a0402515ebf7623ad308e7fde7df31c30f
Type: REGULAR
Attributes:
  FileName=bcc8a209-5788-4f94-922e-cf59fa4a5d66
  Timestamp=1664763100 (2022-10-03 02:11:40 +0000 UTC)
Split Header Signature:
  public key: 03155db91ba2026a8435d0b6b1b0e1d1b52fc8008b203202361c55028922b3e595
  signature: 04dd770a4d1e73e6fc192d81282983d61bd48fbda0e0f740ef6dca72e8caf74d824142ce9810293db98efcc48edfbee5ed0d48568afab31bbe1dab146ea36dc335

02:13:06 [INFO]  [92mExecuting command: neofs-cli --config /tests/neofs_cli_configs/empty_passwd.yml object get --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8081' --wallet '/tests/TemporaryDir/860d6d63-e60a-46e6-bb1d-3d3f07b39dad.json' --cid 'FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa' --oid 'E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU' --file 'TemporaryDir/3685a795-e621-4a63-accd-5ef9bb77c7d5' --no-progress [0m
02:13:06 [INFO]  [92mOutput: [TemporaryDir/3685a795-e621-4a63-accd-5ef9bb77c7d5] Object successfully saved
ID: E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU
CID: FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa
Owner: NPZAC6w7GehsCkii6g9pfuxhYpioLC3xSe
CreatedAt: 19
Size: 2000
HomoHash: 3296263dd3c7323b50cde1add2a1506d555323862ac2a5d9a3667bed8084775435a2abb6ac688ef2953dc34a22cfd72759abcdb0d4576e495859625f49187cd3
Checksum: 7f4dc5c06f7c6bf86b939cc86875d7a0402515ebf7623ad308e7fde7df31c30f
Type: REGULAR
Attributes:
  FileName=bcc8a209-5788-4f94-922e-cf59fa4a5d66
  Timestamp=1664763100 (2022-10-03 02:11:40 +0000 UTC)
Split Header Signature:
  public key: 03155db91ba2026a8435d0b6b1b0e1d1b52fc8008b203202361c55028922b3e595
  signature: 04dd770a4d1e73e6fc192d81282983d61bd48fbda0e0f740ef6dca72e8caf74d824142ce9810293db98efcc48edfbee5ed0d48568afab31bbe1dab146ea36dc335

02:14:21 [INFO]  [92mExecuting command: neofs-cli --config /tests/neofs_cli_configs/empty_passwd.yml object get --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8083' --wallet '/tests/TemporaryDir/860d6d63-e60a-46e6-bb1d-3d3f07b39dad.json' --cid 'FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa' --oid 'E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU' --file 'TemporaryDir/0eed8f4b-0fd4-4c8b-bbbe-1e3b13d3751c' --no-progress [0m
02:14:22 [INFO]  [92mOutput: [TemporaryDir/0eed8f4b-0fd4-4c8b-bbbe-1e3b13d3751c] Object successfully saved
ID: E6hmt3sgQQHWcpeQiqFPD7783ZYdin8viRXXhNsDpfPU
CID: FFyBTXv59CVu7b2HjZYQo5ANx23hP3mUPpt4ys5wqBVa
Owner: NPZAC6w7GehsCkii6g9pfuxhYpioLC3xSe
CreatedAt: 19
Size: 2000
HomoHash: 3296263dd3c7323b50cde1add2a1506d555323862ac2a5d9a3667bed8084775435a2abb6ac688ef2953dc34a22cfd72759abcdb0d4576e495859625f49187cd3
Checksum: 7f4dc5c06f7c6bf86b939cc86875d7a0402515ebf7623ad308e7fde7df31c30f
Type: REGULAR
Attributes:
  FileName=bcc8a209-5788-4f94-922e-cf59fa4a5d66
  Timestamp=1664763100 (2022-10-03 02:11:40 +0000 UTC)
Split Header Signature:
  public key: 03155db91ba2026a8435d0b6b1b0e1d1b52fc8008b203202361c55028922b3e595
  signature: 04dd770a4d1e73e6fc192d81282983d61bd48fbda0e0f740ef6dca72e8caf74d824142ce9810293db98efcc48edfbee5ed0d48568afab31bbe1dab146ea36dc335
```

## Your Environment
**Versions:**

```
neo-go - 0.99.3-2-gaa57779d
neofs-authmate - v0.24.0-7-g3a7ed822
neofs-cli - NeoFS CLI Version v0.32.0-33-g1edc0488
GoVersion - go1.18.4
```

**Environment**
DevEnv

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1829

Signed-off-by: Evgenii Stratonikov <evgeniy@morphbits.ru>

### CODE REVIEW

File: pkg/local_object_storage/engine/delete.go [hunk 1]
```diff
@@ -61,21 +62,29 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
...
 		if err != nil {
-			_, ok := err.(*objectSDK.SplitInfoError)
-			if ok || shard.IsErrRemoved(err) || shard.IsErrObjectExpired(err) {
+			if shard.IsErrRemoved(err) || shard.IsErrObjectExpired(err) {
 				return true
 			}
-			if !shard.IsErrNotFound(err) {
-				e.reportShardError(sh, "could not check object existence", err)
+
+			splitErr, ok := err.(*objectSDK.SplitInfoError)
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/engine/delete.go:80]: Why not `errors.As`?

File: pkg/local_object_storage/engine/delete.go [hunk 2]
```diff
@@ -102,5 +111,46 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
...
+	var fs objectSDK.SearchFilters
+	fs.AddSplitIDFilter(objectSDK.MatchStringEqual, splitID)
+
+	var shPrm shard.InhumePrm
+	if force {
+		shPrm.ForceRemoval()
+	}
+
+	e.iterateOverSortedShards(addr, func(_ int, sh hashedShard) (stop bool) {
+		var prm shard.SelectPrm
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/engine/delete.go:131]: Maybe init it once outside same as `shPrm`?

Comment (fyrchik) [reply pkg/local_object_storage/engine/delete.go:131]: FIxed

File: pkg/local_object_storage/engine/delete.go (hunk 1 above)
Comment (fyrchik) [reply pkg/local_object_storage/engine/delete.go:80]: Mostly because it is faster and was already done like this

Comment (cthulhu-rider) [reply pkg/local_object_storage/engine/delete.go:80]: Oh yeah, missed that it remains unchanged.

File: pkg/local_object_storage/engine/delete.go (hunk 2 above)
Comment (cthulhu-rider) [reply pkg/local_object_storage/engine/delete.go]: `addr` seems static, too.

Comment (fyrchik) [reply pkg/local_object_storage/engine/delete.go]: Fixed

**Review by acid-ant — APPROVED** [review]

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**